.github/workflows/functional-test.yaml

```diff
@@ -165,7 +165,11 @@ jobs:
         id: gen-id
         run: |
           BASE_STR="RADIUS|${GITHUB_SHA}|${GITHUB_SERVER_URL}|${GITHUB_REPOSITORY}|${GITHUB_RUN_ID}|${GITHUB_RUN_ATTEMPT}"
-          UNIQUE_ID=$(echo $BASE_STR | sha1sum | head -c 10)
+          if [ "$GITHUB_EVENT_NAME" == "schedule" ]; then
+            # Add run number to randomize unique id for scheduled runs.
+            BASE_STR="${GITHUB_RUN_NUMBER}|${BASE_STR}"
```

Comment:
is there any reason we only do this for scheduled runs? is there a downside to adding the run number on all runs?

Reply:
If we use GITHUB_RUN_NUMBER for PR event, workflow will create resource groups and resources whenever we trigger functional tests. This results in unnecessary resource creation in our subscription. To avoid this issue and save the cost, we do not use GITHUB_RUN_NUMBER for every event.

Reply:
Even though we have purge workflow, it cleans up only 3 days+ old resources. So we still need to pay 3 days resoruce usage. so it is better not to use `GITHUB_RUN_NUMBER` for every event.